tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,297 @@
+			version := version
+
+			Context(fmt.Sprintf("version %s", version), func() {
+				BeforeAll(func() {
+					Expect(kubectl.CreateNamespace(controlPlaneNamespace)).To(Succeed(), "Namespace failed to be created")
+				})
+
+				When("the Istio CR is created", func() {
+					BeforeAll(func() {
+						istioYAML := `
+apiVersion: operator.istio.io/v1alpha1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: %s
+  namespace: %s`
+						istioYAML = fmt.Sprintf(istioYAML, version, controlPlaneNamespace)
+						fmt.Printf("Istio CR YAML: %s\n", istioYAML)
+						Expect(kubectl.ApplyString(istioYAML)).
+							To(Succeed(), "Istio CR failed to be created")
+						Success("Istio CR created")
+					})
+
+					It("updates the Istio CR status to Reconciled", func() {
+						Eventually(kubectl.GetConditions).
+							WithArguments(controlPlaneNamespace, "istio", istioName).
+							Should(ContainElement(resourceReconciled), "Istio is not Reconciled; unexpected Condition")
+						Success("Istio CR is Reconciled")
+					})
+
+					It("updates the Istio CR status to Ready", func() {
+						Eventually(kubectl.GetConditions).
+							WithArguments(controlPlaneNamespace, "istio", istioName).
+							Should(ContainElement(resourceReady), "Istio is not Ready; unexpected Condition")
+						Success("Istio CR is Ready")
+					})
+
+					It("deploys istiod", func() {
+						Expect(kubectl.GetDeployments(controlPlaneNamespace)).
+							To(Equal([]string{"istiod"}), "Istiod deployment is not present; unexpected list of deployments")
+						Success("Istiod deployment is present")
+
+						// TODO: we need to add a function to get the istio version from the control panel directly
+						// and compare it with the applied version
+						// This is a TODO because actual version.yaml contains for example latest and not the actual version
+						// Posible solution is to add actual version to the version.yaml
+					})
+
+					It("deploys the CNI DaemonSet when running on OpenShift", func() {
+						if ocp {
+							Eventually(kubectl.GetDaemonSets).
+								WithArguments(namespace).
+								Should(ContainElement("istio-cni-node"), "CNI DaemonSet is not deployed; expected list to contain element")
+
+							Eventually(func(g Gomega) {
+								numberAvailable, err := kubectl.GetDaemonSetStatusField(namespace, "istio-cni-node", "numberAvailable")
+								g.Expect(err).ToNot(HaveOccurred(), "Error getting numberAvailable field from istio-cni-node DaemonSet")
+								currentNumberScheduled, err := kubectl.GetDaemonSetStatusField(namespace, "istio-cni-node", "currentNumberScheduled")
+								g.Expect(err).ToNot(HaveOccurred(), "Error getting currentNumberScheduled field from istio-cni-node DaemonSet")
+								g.Expect(numberAvailable).To(Equal(currentNumberScheduled), "CNI DaemonSet Pods are not Available; expected numberAvailable to be equal to currentNumberScheduled")
+							}).Should(Succeed(), "CNI DaemonSet Pods are not Available")
```

Comment:
Ugh. Not pretty, but since this is temporary, I'm okay with merging this as is. I think it would be much cleaner if you just got the entire DaemonSet resource and then compared `status.numberAvailable` and `status.currentNumberScheduled`. But I guess this will be simpler to do when we start using the go client for assertions.

Reply:
Yes, I'll start to work in use the go client as soon as possible this will simplify a lot the assertions and make the code cleaner